ec2/volumes_test.go

```diff
@@ -223,6 +223,37 @@ func (s *ServerTests) TestVolumes(c *C) {
 	assertVolume(c, list.Volumes[0], id2, vol2.VolumeType, vol2.AvailZone, 101, vol2.IOPS)
 }
 
+func (s *ServerTests) TestVolumesTagFilter(c *C) {
+	createVolume := func() string {
+		resp, err := s.ec2.CreateVolume(ec2.CreateVolume{
+			AvailZone:  "us-east-1b",
+			VolumeType: "standard",
+			VolumeSize: 1,
+		})
+		c.Assert(err, IsNil)
+		return resp.Id
+	}
+	destroyVolume := func(id string) {
+		_, err := s.ec2.DeleteVolume(id)
+		c.Assert(err, IsNil)
+	}
+
+	volId1 := createVolume()
+	defer destroyVolume(volId1)
+	volId2 := createVolume()
+	defer destroyVolume(volId2)
+
+	_, err := s.ec2.CreateTags([]string{volId1}, []ec2.Tag{{"key", "value"}})
+	c.Assert(err, IsNil)
+
+	filter := ec2.NewFilter()
+	filter.Add("tag:key", "value")
+	resp, err := s.ec2.Volumes(nil, filter)
```

Comment:
maybe also a check that the volume fails to match the filter before the tags are added